spec/unit/config_files_inspector_spec.rb

```diff
@@ -96,231 +118,349 @@
+          apache_config_1, apache_config_2, apache_config_3, apache_config_4, apache_config_5
+        ]
+        @expected_data = ConfigFilesScope.new(
+          extracted: false,
+          files: ConfigFileList.new(
+            [
+              apache_config_1, apache_config_2, iscsi_config_1, apache_config_3,
+              apache_config_4, apache_config_5
+            ]
+          )
+        )
+
+        allow_any_instance_of(ConfigFilesInspector).to receive(
+          :packages_with_config_files
+        ).and_return(["apache2-2.4.6", "open-iscsi-2.0.873"])
+        allow_any_instance_of(ConfigFilesInspector).to receive(:config_file_changes).with(
+          "apache2-2.4.6"
+        ).and_return(expected_apache_config_file_data)
+        allow_any_instance_of(ConfigFilesInspector).to receive(:config_file_changes).with(
+          "open-iscsi-2.0.873"
+        ).and_return(iscsi_config_1)
+        allow_any_instance_of(ConfigFilesInspector).to receive(:check_requirements)
+      end
+
+      it "returns data about modified config files when requirements are fulfilled" do
+        expect_data_gather_cmds(
+          system,
+          config_paths, stat_output
+        )
+
+        inspector = ConfigFilesInspector.new(system, description)
+        inspector.inspect(filter)
+
+        expect(description["config_files"]).to eq(@expected_data)
+        expect(inspector.summary).to include("6 changed configuration files")
+      end
+
+      it "returns empty when no modified config files are there" do
+        allow_any_instance_of(ConfigFilesInspector).to receive(
+          :packages_with_config_files
+        ).and_return([])
+
+        inspector = ConfigFilesInspector.new(system, description)
+        inspector.inspect(filter)
+        expected = ConfigFilesScope.new(
+          extracted: false,
+          files: ConfigFileList.new
+        )
+        expect(description["config_files"]).to eq(expected)
+      end
 
-    it "returns sorted data" do
-      system = double
-      expect_inspect_configfiles(system, true)
+      it "raise an error when requirements are not fulfilled" do
+        allow_any_instance_of(ConfigFilesInspector).to receive(
+          :check_requirements
+        ).and_raise(Machinery::Errors::MissingRequirement)
 
-      inspector = ConfigFilesInspector.new(system, description)
-      inspector.inspect(filter, :extract_changed_config_files => true)
-      names = description["config_files"].files.map(&:name)
+        inspector = ConfigFilesInspector.new(system, description)
+        expect { inspector.inspect(filter) }.to raise_error(
+          Machinery::Errors::MissingRequirement
+        )
+      end
+
+      it "extracts changed configuration files" do
+        cfdir = File.join(store.description_path(name), "config_files")
+        expect(system).to receive(:retrieve_files).with(
+          config_paths,
+          cfdir
+        )
+        expect_data_gather_cmds(
```

Comment:
Thanks for the feedback.
Andre explained me in this case it is needed, because system is not a real system but a double and that both stubs are on the same level but retrieve_files is written in the text and expect_data_gather_cmds is extracted to the top of the file.